cmd/server/app/watch.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright © 2021 Dan Lorenc <lorenc.d@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/google/trillian"
+	tclient "github.com/google/trillian/client"
+	tcrypto "github.com/google/trillian/crypto"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+	"github.com/google/trillian/types"
+
+	"github.com/sigstore/rekor/cmd/cli/app"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/log"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/fileblob"
+	_ "gocloud.dev/blob/gcsblob"
+)
+
+// watchCmd represents the serve command
+var watchCmd = &cobra.Command{
+	Use:   "watch",
+	Short: "start a process to watch and record STH's from Rekor",
+	Long:  `start a process to watch and record STH's from Rekor`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		// these are bound here so that they are not overwritten by other commands
+		if err := viper.BindPFlags(cmd.Flags()); err != nil {
+			log.Logger.Fatal("Error initializing cmd line args: ", err)
+		}
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// Setup the logger to dev/prod
+		log.ConfigureLogger(viper.GetString("log_type"))
+
+		// workaround for https://github.com/sigstore/rekor/issues/68
+		// from https://github.com/golang/glog/commit/fca8c8854093a154ff1eb580aae10276ad6b1b5f
+		_ = flag.CommandLine.Parse([]string{})
+
+		host := viper.GetString("rekor_server.address")
+		port := viper.GetUint("rekor_server.port")
+		interval := viper.GetDuration("interval")
+		url := fmt.Sprintf("http://%s:%d", host, port)
+		c, err := app.GetRekorClient(url)
+		if err != nil {
+			return err
+		}
+
+		keyResp, err := c.Tlog.GetPublicKey(nil)
+		if err != nil {
+			return err
+		}
+		publicKey := keyResp.Payload
+		block, _ := pem.Decode([]byte(publicKey))
+		if block == nil {
+			return errors.New("failed to decode public key of server")
+		}
+
+		pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+		if err != nil {
+			return err
+		}
+
+		verifier := tclient.NewLogVerifier(hasher.DefaultHasher, pub, crypto.SHA256)
+		ctx := context.Background()
+		bucketURL := os.Getenv("REKOR_STH_BUCKET")
```

Comment:
The next function call will catch this, but we may want to consider fatally logging a specific error message if this env var is unset.

Reply:
Done!